hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/GroupP.java

```diff
@@ -82,4 +81,26 @@ protected boolean tryProcess(int ordinal, @Nonnull Object item) {
     public boolean complete() {
         return emitFromTraverser(resultTraverser);
     }
+
+    private class ResultTraverser implements Traverser<OUT> {
+        private final BiFunction<? super K, ? super R, OUT> mapToOutputFn;
+        private Iterator<Entry<K, A>> iter;
+
+        ResultTraverser(BiFunction<? super K, ? super R, OUT> mapToOutputFn) {
+            this.mapToOutputFn = mapToOutputFn;
+        }
+
+        @Override
+        public OUT next() {
+            if (iter == null) {
```

Comment:
We can move this to the constructor and construct the `resultTraverser` in `complete`, if it's null. This would be less checking.